crates/services/consensus_module/poa/src/service_test/trigger_tests.rs

```diff
@@ -299,3 +317,144 @@ async fn interval_trigger_doesnt_react_to_full_txpool() -> anyhow::Result<()> {
+) {
+    // Given
+
+    let block_time = Duration::from_secs(10);
+    let second_block_delay = Duration::from_secs(5);
+    let offset = Duration::from_secs(1);
+    let mut ctx = DefaultContext::new(Config {
+        trigger: Trigger::Interval { block_time },
+        signer: SignMode::Key(test_signing_key()),
+        metrics: false,
+        ..Default::default()
+    });
+    ctx.status_sender.send_replace(Some(TxId::zeroed()));
+    let start_time = ctx.test_ctx.time.watch().now();
+
+    // When
+
+    // We produce the first block in real time.
+    time::sleep(block_time + offset).await;
+    ctx.test_ctx.time.advance_with_tokio();
+    let first_block_time = ctx.block_import.try_recv().unwrap().entity.header().time();
+
+    // But we produce second block with a delay relative to real time.
+    ctx.test_ctx.time.advance(block_time + second_block_delay);
+    time::sleep(block_time).await;
+    let second_block_time = ctx.block_import.try_recv().unwrap().entity.header().time();
+
+    // And the third block is produced without advancing real time.
+    time::sleep(block_time).await;
+    let third_block_time = ctx.block_import.try_recv().unwrap().entity.header().time();
+
+    // Then
+
+    // We should only have produced the three blocks.
+    assert!(matches!(
+        ctx.block_import.try_recv(),
+        Err(broadcast::error::TryRecvError::Empty)
+    ));
+
+    // The fist block should be produced after the given block time.
+    assert_eq!(first_block_time, start_time + block_time.as_secs());
+
+    // The second block should have a delay in its timestamp.
+    assert_eq!(
+        second_block_time,
+        first_block_time
+            + block_time.as_secs()
+            + offset.as_secs()
+            + second_block_delay.as_secs()
+    );
+
+    // The third block should be produced `block_time` in the future relative to the second block time.
+    assert_eq!(third_block_time, second_block_time + block_time.as_secs());
+
+    ctx.test_ctx.service.stop_and_await().await.unwrap();
+}
+
+#[tokio::test]
+async fn interval_trigger_produces_blocks_in_the_future_when_time_rewinds() {
+    // Given
+
+    let block_time = Duration::from_secs(10);
+    let offset = Duration::from_secs(1);
+    let mut ctx = DefaultContext::new(Config {
+        trigger: Trigger::Interval { block_time },
+        signer: SignMode::Key(test_signing_key()),
+        metrics: false,
+        ..Default::default()
+    });
+    ctx.status_sender.send_replace(Some(TxId::zeroed()));
+    let start_time = ctx.test_ctx.time.watch().now();
+
+    // When
+
+    // We produce the first block in real time.
+    time::sleep(block_time + offset).await;
+    ctx.test_ctx.time.advance_with_tokio();
+    let first_block_time = ctx.block_import.try_recv().unwrap().entity.header().time();
+
+    // And we rewind time before attempting to produce the next block.
+    ctx.test_ctx.time.rewind(block_time);
```

Comment:
`ctx` should just provide a method for this IMO. 

Either that, or we have a separate handle for manipulating time.

Reply:
Sure, I can add a convenience function here.

Reply:
Done.